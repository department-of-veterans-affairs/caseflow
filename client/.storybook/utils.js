module.exports = {
  TASKS_ARRAY: [
    {
      "id": "17397",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithzieme",
          "relationship": "Bob Smithzieme"
        },
        "appeal": {
          "id": "6115",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Caregiver | Eligibility",
        "tasks_url": "/decision_reviews/vha",
        "id": 17397,
        "created_at": "2022-12-28T13:19:13.801-05:00",
        "veteran_participant_id": "794566840",
        "veteran_ssn": "481174543",
        "assigned_on": "2022-12-28T13:19:13.771-05:00",
        "assigned_at": "2022-12-28T13:19:13.771-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "17395",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Cathy Smith",
          "relationship": "Cathy Smith"
        },
        "appeal": {
          "id": "6114",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Initial Eligibility and Enrollment in VHA Healthcare",
        "tasks_url": "/decision_reviews/vha",
        "id": 17395,
        "created_at": "2022-12-22T12:03:57.796-05:00",
        "veteran_participant_id": "794566954",
        "veteran_ssn": "941308343",
        "assigned_on": "2022-12-22T12:03:57.795-05:00",
        "assigned_at": "2022-12-22T12:03:57.795-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "17394",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Tom Brady",
          "relationship": "Tom Brady"
        },
        "appeal": {
          "id": "17",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Initial Eligibility and Enrollment in VHA Healthcare",
        "tasks_url": "/decision_reviews/vha",
        "id": 17394,
        "created_at": "2022-12-22T11:46:24.041-05:00",
        "veteran_participant_id": "794566900",
        "veteran_ssn": "333089389",
        "assigned_on": "2022-12-22T11:46:24.017-05:00",
        "assigned_at": "2022-12-22T11:46:24.017-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Supplemental Claim",
        "business_line": "vha"
      }
    },
    {
      "id": "17386",
      "type": "board_grant_effectuation_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithtremblay",
          "relationship": "Bob Smithtremblay"
        },
        "appeal": {
          "id": "3f67e4b2-392b-4248-a722-0845b2d3a29e",
          "isLegacyAppeal": false,
          "issueCount": 0,
          "activeOrDecidedRequestIssues": 0
        },
        "issue_count": 0,
        "issue_types": "",
        "tasks_url": "/decision_reviews/vha",
        "id": 17386,
        "created_at": "2022-12-16T10:47:58.296-05:00",
        "veteran_participant_id": "794480017",
        "veteran_ssn": "537817892",
        "assigned_on": "2022-12-16T10:47:54.321-05:00",
        "assigned_at": "2022-12-16T10:47:54.321-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Board Grant",
        "business_line": "vha"
      }
    },
    {
      "id": "17376",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Org Other Claimant ",
          "relationship": "Org Other Claimant "
        },
        "appeal": {
          "id": "14",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Caregiver | Revocation/Discharge",
        "tasks_url": "/decision_reviews/vha",
        "id": 17376,
        "created_at": "2022-12-07T10:39:35.204-05:00",
        "veteran_participant_id": "794566900",
        "veteran_ssn": "476407820",
        "assigned_on": "2022-12-07T10:39:35.175-05:00",
        "assigned_at": "2022-12-07T10:39:35.175-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Supplemental Claim",
        "business_line": "vha"
      }
    },
    {
      "id": "17375",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Child Person",
          "relationship": "Child Person"
        },
        "appeal": {
          "id": "13",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Initial Eligibility and Enrollment in VHA Healthcare",
        "tasks_url": "/decision_reviews/vha",
        "id": 17375,
        "created_at": "2022-12-07T10:32:31.051-05:00",
        "veteran_participant_id": "794566816",
        "veteran_ssn": "682838204",
        "assigned_on": "2022-12-07T10:32:31.026-05:00",
        "assigned_at": "2022-12-07T10:32:31.026-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Supplemental Claim",
        "business_line": "vha"
      }
    },
    {
      "id": "15128",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithmckenzie",
          "relationship": "Bob Smithmckenzie"
        },
        "appeal": {
          "id": "5061",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Initial Eligibility and Enrollment in VHA Healthcare",
        "tasks_url": "/decision_reviews/vha",
        "id": 15128,
        "created_at": "2022-11-18T16:43:16.547-05:00",
        "veteran_participant_id": "880774999",
        "veteran_ssn": "223450678",
        "assigned_on": "2022-11-29T15:25:38.733-05:00",
        "assigned_at": "2022-11-29T15:25:38.733-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "17382",
      "type": "veteran_record_request",
      "attributes": {
        "claimant": {
          "name": "Bob Smithherzog",
          "relationship": "Bob Smithherzog"
        },
        "appeal": {
          "id": "2f13f0cd-0262-4ec8-87d1-a849fa0a7628",
          "isLegacyAppeal": false,
          "issueCount": 1
        },
        "issue_count": 1,
        "issue_types": "Spina Bifida Treatment (Non-Compensation)",
        "tasks_url": "/decision_reviews/vha",
        "id": 17382,
        "created_at": "2022-12-14T13:55:26.852-05:00",
        "veteran_participant_id": "104410003",
        "veteran_ssn": "408793474",
        "assigned_on": "2022-11-14T13:55:26.762-05:00",
        "assigned_at": "2022-11-14T13:55:26.762-05:00",
        "closed_at": null,
        "started_at": null,
        "type": "Record Request",
        "business_line": "vha"
      }
    },
    {
      "id": "16359",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithpacocha",
          "relationship": "Bob Smithpacocha"
        },
        "appeal": {
          "id": "5105",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Foreign Medical Program",
        "tasks_url": "/decision_reviews/vha",
        "id": 16359,
        "created_at": "2022-11-30T10:23:06.122-05:00",
        "veteran_participant_id": "982170028",
        "veteran_ssn": "485696446",
        "assigned_on": "2022-10-31T11:23:06.066-04:00",
        "assigned_at": "2022-10-31T11:23:06.066-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16350",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithlangworth",
          "relationship": "Bob Smithlangworth"
        },
        "appeal": {
          "id": "5096",
          "isLegacyAppeal": false,
          "issueCount": 2,
          "activeOrDecidedRequestIssues": 2
        },
        "issue_count": 2,
        "issue_types": "Foreign Medical Program,Spina Bifida Treatment (Non-Compensation)",
        "tasks_url": "/decision_reviews/vha",
        "id": 16350,
        "created_at": "2022-11-30T10:23:05.591-05:00",
        "veteran_participant_id": "982170019",
        "veteran_ssn": "343776327",
        "assigned_on": "2022-10-31T11:23:05.542-04:00",
        "assigned_at": "2022-10-31T11:23:05.542-04:00",
        "closed_at": null,
        "started_at": "2022-12-22T11:31:33.285-05:00",
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16337",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithschowalter",
          "relationship": "Bob Smithschowalter"
        },
        "appeal": {
          "id": "5083",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Spina Bifida Treatment (Non-Compensation)",
        "tasks_url": "/decision_reviews/vha",
        "id": 16337,
        "created_at": "2022-11-30T10:23:04.813-05:00",
        "veteran_participant_id": "982170006",
        "veteran_ssn": "479749554",
        "assigned_on": "2022-10-31T11:23:04.775-04:00",
        "assigned_at": "2022-10-31T11:23:04.775-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16335",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithmcdermott",
          "relationship": "Bob Smithmcdermott"
        },
        "appeal": {
          "id": "5081",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Camp Lejune Family Member",
        "tasks_url": "/decision_reviews/vha",
        "id": 16335,
        "created_at": "2022-11-30T10:23:04.712-05:00",
        "veteran_participant_id": "982170004",
        "veteran_ssn": "517398195",
        "assigned_on": "2022-10-31T11:23:04.675-04:00",
        "assigned_at": "2022-10-31T11:23:04.675-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16360",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithbraun",
          "relationship": "Bob Smithbraun"
        },
        "appeal": {
          "id": "5106",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Camp Lejune Family Member",
        "tasks_url": "/decision_reviews/vha",
        "id": 16360,
        "created_at": "2022-11-30T10:23:06.190-05:00",
        "veteran_participant_id": "982170029",
        "veteran_ssn": "494430453",
        "assigned_on": "2022-10-30T11:23:06.144-04:00",
        "assigned_at": "2022-10-30T11:23:06.144-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16355",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithwillms",
          "relationship": "Bob Smithwillms"
        },
        "appeal": {
          "id": "5101",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Other",
        "tasks_url": "/decision_reviews/vha",
        "id": 16355,
        "created_at": "2022-11-30T10:23:05.856-05:00",
        "veteran_participant_id": "982170024",
        "veteran_ssn": "367801178",
        "assigned_on": "2022-10-30T11:23:05.817-04:00",
        "assigned_at": "2022-10-30T11:23:05.817-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    },
    {
      "id": "16348",
      "type": "decision_review_task",
      "attributes": {
        "claimant": {
          "name": "Bob Smithfranecki",
          "relationship": "Bob Smithfranecki"
        },
        "appeal": {
          "id": "5094",
          "isLegacyAppeal": false,
          "issueCount": 1,
          "activeOrDecidedRequestIssues": 1
        },
        "issue_count": 1,
        "issue_types": "Caregiver | Other",
        "tasks_url": "/decision_reviews/vha",
        "id": 16348,
        "created_at": "2022-11-30T10:23:05.473-05:00",
        "veteran_participant_id": "982170017",
        "veteran_ssn": "602779430",
        "assigned_on": "2022-10-30T11:23:05.433-04:00",
        "assigned_at": "2022-10-30T11:23:05.433-04:00",
        "closed_at": null,
        "started_at": null,
        "type": "Higher-Level Review",
        "business_line": "vha"
      }
    }
  ]
};
