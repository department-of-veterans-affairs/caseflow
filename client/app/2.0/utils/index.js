/* eslint-disable  */
export * from 'utils/reader/index';

/* eslint-enable  */
