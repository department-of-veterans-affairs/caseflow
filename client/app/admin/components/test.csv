VET_LAST_NAME,VET_FIRST_NAME,VET_MIDDLE_NAME,VET_DATE_OF_BIRTH,VET_SSN,VET_PARTICIPANT_ID,VET_FILE_NUMBER,APPELLANT_LAST_NAME,APPELLANT_FIRST_NAME,APPELLANT_MIDDLE_NAME,APPELLANT_DATE_OF_BIRTH,APPELLANT_SSN,APPELLANT_GENDER,APPELLANT_ADDRESS,APPELLANT_PHONE,APPELLANT_EMAIL,APPELLANT_EDI_PI,APPELLANT_CORP_PID,APPELLANT_VACOLS_INTERNAL_ID,RELATIONSHIP_TO_VETERAN