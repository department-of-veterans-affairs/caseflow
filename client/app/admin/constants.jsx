export const ACTIONS = {
  SET_FEATURE_TOGGLES: 'SET_FEATURE_TOGGLES',
  POST_VETERAN_EXTRACT: 'POST_VETERAN_EXTRACT',
  STARTED_VETERAN_EXTRACT: 'STARTED_VETERAN_EXTRACT',
  FAILED_VETERAN_EXTRACT: 'FAILED_VETERAN_EXTRACT'
};
