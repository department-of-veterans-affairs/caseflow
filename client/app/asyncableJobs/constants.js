export const ACTIONS = {
  MANUAL_JOB_STARTED: 'MANUAL_JOB_STARTED',
  MANUAL_JOB_FAILED: 'MANUAL_JOB_FAILED',
  CLEAR_MANUAL_JOB_TYPE: 'CLEAR_MANUAL_JOB_TYPE',
};