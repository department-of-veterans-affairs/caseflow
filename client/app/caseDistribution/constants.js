import DISTRIBUTION from '../../constants/DISTRIBUTION';

const LEVERS = 'levers';

export const Constant = {
  LEVERS
};

export const sectionTitles = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]: DISTRIBUTION.ama_hearings_section_title,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]: DISTRIBUTION.ama_direct_review_section_title,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]: DISTRIBUTION.ama_evidence_submission_section_title
};

export const docketTimeGoalPriorMappings = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_hearings_docket_time_goals,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_direct_review_docket_time_goals,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_evidence_submission_docket_time_goals
};
