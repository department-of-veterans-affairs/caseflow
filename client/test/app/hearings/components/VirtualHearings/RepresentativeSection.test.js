import React from 'react';
import { mount } from 'enzyme';

import { virtualHearing, defaultHearing } from 'test/data/hearings';
import { HEARING_CONVERSION_TYPES } from 'app/hearings/constants';
import { RepresentativeSection } from 'app/hearings/components/VirtualHearings/RepresentativeSection';
import { amaHearing } from 'test/data';
import { VirtualHearingSection } from 'app/hearings/components/VirtualHearings/Section';
import { AddressLine } from 'app/hearings/components/details/Address';
import { VirtualHearingEmail } from 'app/hearings/components/VirtualHearings/Emails';
import { Timezone } from 'app/hearings/components/VirtualHearings/Timezone';
import { ReadOnly } from 'app/hearings/components/details/ReadOnly';
import { getAppellantTitle } from 'app/hearings/utils';
import TextField from 'app/components/TextField';

const updateSpy = jest.fn();

describe('RepresentativeSection', () => {
  test('Matches snapshot with default props', () => {
    // Run the test
    const representativeSection = mount(
      <RepresentativeSection
        appellantTitle="Veteran"
        virtualHearing={virtualHearing.virtualHearing}
        hearing={defaultHearing}
        type={HEARING_CONVERSION_TYPES[0]}
        update={updateSpy}
      />
    );

    // Assertions
    expect(representativeSection.find(AddressLine)).toHaveLength(1);
    expect(representativeSection.find(VirtualHearingEmail)).toHaveLength(1);
    expect(representativeSection.find(TextField)).toHaveLength(1);
    expect(representativeSection).toMatchSnapshot();
  });

  test('Does not allow editing emails when read-only', () => {
    // Run the test
    const representativeSection = mount(
      <RepresentativeSection
        readOnly
        appellantTitle="Veteran"
        virtualHearing={virtualHearing.virtualHearing}
        hearing={defaultHearing}
        type={HEARING_CONVERSION_TYPES[0]}
        update={updateSpy}
      />
    );

    // Ensure the emails are read-only
    expect(representativeSection.find(TextField)).toHaveLength(0);
    representativeSection.find(VirtualHearingEmail).map((node) => expect(node.prop('readOnly')).toEqual(true));
    expect(representativeSection).toMatchSnapshot();
  });

  test('Displays timezone when virtual', () => {
    // Run the test
    const representativeSection = mount(
      <RepresentativeSection
        virtual
        appellantTitle="Veteran"
        virtualHearing={virtualHearing.virtualHearing}
        hearing={defaultHearing}
        type={HEARING_CONVERSION_TYPES[0]}
        update={updateSpy}
      />
    );

    // Ensure the emails are read-only
    expect(representativeSection.find(Timezone)).toHaveLength(1);
    expect(representativeSection).toMatchSnapshot();
  });

  test('Shows Representative not present message when no representative', () => {
    const representativeSection = mount(
      <RepresentativeSection
        appellantTitle="Veteran"
        virtualHearing={virtualHearing.virtualHearing}
        hearing={{
          ...amaHearing,
          representative: null
        }}
        type={HEARING_CONVERSION_TYPES[0]}
        update={updateSpy}
      />
    );

    // Assertions
    expect(representativeSection.find(AddressLine)).toHaveLength(0);
    expect(representativeSection.find(VirtualHearingEmail)).toHaveLength(1);
    expect(representativeSection.find(VirtualHearingSection).first().
      find(ReadOnly).
      prop('text')).toEqual(
      `The ${getAppellantTitle(amaHearing.appellantIsNotVeteran)} does not have a representative recorded in VBMS`
    );
    expect(representativeSection).toMatchSnapshot();
  });

  test('Shows Representative name when representative address blank', () => {
    const representativeSection = mount(
      <RepresentativeSection
        appellantTitle="Veteran"
        virtualHearing={virtualHearing.virtualHearing}
        hearing={{
          ...amaHearing,
          representativeAddress: null
        }}
        type={HEARING_CONVERSION_TYPES[0]}
        update={updateSpy}
      />
    );

    // Assertions
    expect(representativeSection.find(AddressLine)).toHaveLength(1);
    expect(representativeSection.find(VirtualHearingEmail)).toHaveLength(1);
    expect(representativeSection.find(AddressLine).first().
      text()).toMatch(amaHearing.representativeName);
    expect(representativeSection).toMatchSnapshot();
  });
});
