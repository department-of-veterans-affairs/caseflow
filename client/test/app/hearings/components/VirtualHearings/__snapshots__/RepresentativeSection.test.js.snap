// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepresentativeSection Displays timezone when showTimezoneField is passed as prop 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <h2>
    Power of Attorney (POA)
  </h2>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      PARALYZED VETERANS OF AMERICA, INC.
9999 MISSION ST
SAN FRANCISCO, CA 94103
    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class=""
      >
        <div
          class="cf-form-dropdown dropdown-representativeTz"
          data-css-1gh7024=""
        >
          <label
            class="question-label"
            for="representative-tz"
            id="representative-tz-label"
          >
            <strong>
              <span>
                POA/Representative Timezone
                <span
                  class="cf-optional"
                >
                  Optional
                </span>
              </span>
            </strong>
          </label>
          <div
            class="cf-select"
          >
            <div
              class=" css-2b097c-container"
            >
              <div
                class="cf-select__control css-yk16xz-control"
              >
                <div
                  class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                >
                  <div
                    class="cf-select__single-value css-1uccc91-singleValue"
                  >
                    Pacific Time (US & Canada) (3:00 AM)
                  </div>
                  <div
                    class="css-81nyic-Input"
                  >
                    <div
                      class="cf-select__input"
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-labelledby="representative-tz-label"
                        aria-owns="representative-tz-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="representative-tz"
                        role="combobox"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="representativeTz"
                type="hidden"
                value="America/Los_Angeles"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="helper-text"
      >
        The hearing time in email notifications is calculated based on the timezone.
      </div>
    </div>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class="cf-form-textinput cf-inline-field   "
      >
        <label
          for="POA/Representative Email (for these notifications only)"
        >
          <strong>
            <span>
              POA/Representative Email (for these notifications only)
              <span
                class="cf-optional"
              >
                Optional
              </span>
            </span>
          </strong>
        </label>
        <div
          class="input-container"
        >
          <input
            class="cf-form-textinput cf-inline-field"
            id="POA/Representative Email (for these notifications only)"
            name="POA/Representative Email (for these notifications only)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="helper-text"
      >
        This email address will be used to send notifications for this hearing only.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RepresentativeSection Does not allow editing emails when read-only 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <h2>
    Power of Attorney (POA)
  </h2>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      PARALYZED VETERANS OF AMERICA, INC.
9999 MISSION ST
SAN FRANCISCO, CA 94103
    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        data-css-1a5irm0=""
      >
        <strong>
          POA/Representative Email (for these notifications only)
        </strong>
        <pre>
          None
        </pre>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RepresentativeSection Does not display address when formFieldsOnly = true 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <strong>
     Power of Attorney (POA)
  </strong>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      PARALYZED VETERANS OF AMERICA, INC.
    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class="cf-form-textinput cf-inline-field   "
      >
        <label
          for="POA/Representative Email (for these notifications only)"
        >
          <strong>
            <span>
              POA/Representative Email (for these notifications only)
              <span
                class="cf-optional"
              >
                Optional
              </span>
            </span>
          </strong>
        </label>
        <div
          class="input-container"
        >
          <input
            class="cf-form-textinput cf-inline-field"
            id="POA/Representative Email (for these notifications only)"
            name="POA/Representative Email (for these notifications only)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="helper-text"
      >
        This email address will be used to send notifications for this hearing only.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RepresentativeSection Matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <h2>
    Power of Attorney (POA)
  </h2>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      PARALYZED VETERANS OF AMERICA, INC.
9999 MISSION ST
SAN FRANCISCO, CA 94103
    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class="cf-form-textinput cf-inline-field   "
      >
        <label
          for="POA/Representative Email (for these notifications only)"
        >
          <strong>
            <span>
              POA/Representative Email (for these notifications only)
              <span
                class="cf-optional"
              >
                Optional
              </span>
            </span>
          </strong>
        </label>
        <div
          class="input-container"
        >
          <input
            class="cf-form-textinput cf-inline-field"
            id="POA/Representative Email (for these notifications only)"
            name="POA/Representative Email (for these notifications only)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="helper-text"
      >
        This email address will be used to send notifications for this hearing only.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RepresentativeSection Shows Representative name when representative address blank 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <h2>
    Power of Attorney (POA)
  </h2>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      PARALYZED VETERANS OF AMERICA, INC.

    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class="cf-form-textinput cf-inline-field   "
      >
        <label
          for="POA/Representative Email (for these notifications only)"
        >
          <strong>
            <span>
              POA/Representative Email (for these notifications only)
              <span
                class="cf-optional"
              >
                Optional
              </span>
            </span>
          </strong>
        </label>
        <div
          class="input-container"
        >
          <input
            class="cf-form-textinput cf-inline-field"
            id="POA/Representative Email (for these notifications only)"
            name="POA/Representative Email (for these notifications only)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="helper-text"
      >
        This email address will be used to send notifications for this hearing only.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RepresentativeSection Shows Representative not present message when no representative 1`] = `
<DocumentFragment>
  <div
    class="cf-help-divider"
  />
  <h2>
    Power of Attorney (POA)
  </h2>
  <div
    data-css-1a5irm0=""
  >
    <pre>
      The Veteran does not have a representative recorded in VBMS
    </pre>
  </div>
  <div
    class="usa-grid css-1jfbwxw"
  >
    <div
      class="usa-width-one-half"
    >
      <div
        class="cf-form-textinput cf-inline-field   "
      >
        <label
          for="POA/Representative Email (for these notifications only)"
        >
          <strong>
            <span>
              POA/Representative Email (for these notifications only)
              <span
                class="cf-optional"
              >
                Optional
              </span>
            </span>
          </strong>
        </label>
        <div
          class="input-container"
        >
          <input
            class="cf-form-textinput cf-inline-field"
            id="POA/Representative Email (for these notifications only)"
            name="POA/Representative Email (for these notifications only)"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="helper-text"
      >
        This email address will be used to send notifications for this hearing only.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
